src/reanimated2/helperTypes.ts

```diff
@@ -80,7 +80,9 @@ type PickStyleProps<T> = Pick<
 >;
 
 type StyleAnimatedProps<P extends object> = {
-  [K in keyof PickStyleProps<P>]: StyleProp<AnimateStyle<P[K]>>;
+  [K in keyof PickStyleProps<P>]: StyleProp<
+    AnimateStyle<P[K]> | AnimateStyle<MaybeSharedValue<P[K]>>
+  >;
```

Comment:
You can move `AnimateStyle<` out like that:
```suggestion
  [K in keyof PickStyleProps<P>]: StyleProp<
    AnimateStyle<P[K] | MaybeSharedValue<P[K]>>
  >;
```